rest/oidc_api_test.go

```diff
@@ -0,0 +1,80 @@
+package rest
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/couchbase/sync_gateway/auth"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHandleOIDCCommon(t *testing.T) {
+	var (
+		wellKnownPath       = "/.well-known/openid-configuration"
+		headerLocation      = "Location"
+		providerNameGoogle  = "google"
+		clientIDGoogle      = "client.id.google"
+		validationKeyGoogle = "validation.key.google"
+	)
+
+	// Mock OpenID Connect Provider Discovery Endpoints
+	var issuerGoogle string
+	mux := http.NewServeMux()
+	mux.HandleFunc("/google"+wellKnownPath, func(res http.ResponseWriter, req *http.Request) {
+		res.Header().Set("Content-Type", "application/json")
+		body := `{
+			"issuer": "${issuer}", 
+			"authorization_endpoint": "${issuer}/auth", 
+        	"token_endpoint": "https://oauth2.googleapis.com/token",
+			"jwks_uri": "https://www.googleapis.com/oauth2/v3/certs", 
+			"id_token_signing_alg_values_supported": ["RS256"]
+		}`
+		_, err := io.WriteString(res, strings.ReplaceAll(body, "${issuer}", issuerGoogle))
+		require.NoError(t, err, "Failed to write response body for google provider")
+	})
+
+	ts := httptest.NewServer(mux)
+	defer ts.Close()
+
+	issuerGoogle = ts.URL + "/google"
+	providerGoogle := auth.OIDCProvider{
+		Name:          providerNameGoogle,
+		Issuer:        issuerGoogle,
+		ClientID:      &clientIDGoogle,
+		ValidationKey: &validationKeyGoogle,
+		DiscoveryURI:  issuerGoogle + wellKnownPath,
+	}
+
+	providerMap := auth.OIDCProviderMap{providerNameGoogle: &providerGoogle}
+	openIDConnectOptions := auth.OIDCOptions{Providers: providerMap, DefaultProvider: &providerGoogle.Name}
```

Comment:
I feel like I've seen this exact same test setup code a lot, is there not a way to reuse it and make it easier to write OIDC tests going forwards?

Is this not what the test provider was intended to be used for?